src/WebView.android.tsx

```diff
@@ -27,6 +27,17 @@ import {
 
 import styles from './WebView.styles';
 
+const { getWebViewDefaultUserAgent } = NativeModules.RNCWebViewUtils;
+
+let userAgentCache: string | undefined
+
+function getUserAgent() {
+  if (typeof userAgentCache === 'undefined') {
+    userAgentCache = getWebViewDefaultUserAgent() || ''
```

Comment:
This is async (returns a promise).

Reply:
@RodrigoEspinosa How can we fix this so that the webview won't be initialized until we get the version?

Reply:
We could add a static initialize method, for example. And block all webview creation attempt until it's completed.

Reply:
You can use `useState` + `useEffect` - but I'd recommend fixing the memoizing first. 

```js
const [webViewVersion, setWebViewVersion] = useState()

useEffect(() => {
  getWebViewDefaultUserAgent().then(setWebViewVersion)
}, [])

if (!webViewVersion) {
  return null // This will stop the rendering until webViewVersion is defined.
}
```

Reply:
But this code already provides memoization